### ISSUE TITLE
使用vllm在init engine后卡住

### ISSUE THREAD

**Issue by lingzehui-123** [OpenRLHF/OpenRLHF#951]
我使用两张A100进行实验，运行至
(LLMRayActor pid=123434) INFO 04-09 01:52:05 [backends.py:425] Dynamo bytecode transform time: 6.67 s
(LLMRayActor pid=123434) INFO 04-09 01:52:05 [backends.py:115] Directly load the compiled graph for shape None from the cache
(LLMRayActor pid=123434) INFO 04-09 01:52:12 [monitor.py:33] torch.compile takes 6.67 s in total
(LLMRayActor pid=123434) INFO 04-09 01:52:12 [kv_cache_utils.py:566] GPU KV cache size: 5,854,768 tokens
(LLMRayActor pid=123434) INFO 04-09 01:52:12 [kv_cache_utils.py:569] Maximum concurrency for 2,048 tokens per request: 2858.77x
(LLMRayActor pid=123434) INFO 04-09 01:52:46 [gpu_model_runner.py:1534] Graph capturing finished in 33 secs, took 1.27 GiB
(LLMRayActor pid=123434) INFO 04-09 01:52:46 [core.py:151] init engine (profile, create kv cache, warmup model) took 47.83 seconds
之后就卡住不动了，此时我的两张显卡其中一张显存几乎占满，另一张空闲
使用脚本为train_ppo_llama_ray.sh
set -x 

ray job submit --address="http://127.0.0.1:8265" \
   --runtime-env-json='{"working_dir": "./openrlhf"}' \
   -- python3 -m openrlhf.cli.train_ppo_ray \
   --ref_num_nodes 1 \
   --ref_num_gpus_per_node 1 \
   --reward_num_nodes 1 \
   --reward_num_gpus_per_node 1 \
   --critic_num_nodes 1 \
   --critic_num_gpus_per_node 1 \
   --actor_num_nodes 1 \
   --actor_num_gpus_per_node 1 \
   --vllm_num_engines 1 \
   --vllm_tensor_parallel_size 1 \
   --colocate_critic_reward \
   --colocate_actor_ref \
   --pretrain './llm/model/Qwen2.5-0.5B-Instruct' \
   --reward_pretrain './llm/model/Qwen2.5-0.5B-Instruct' \
   --save_path /openrlhf/examples/checkpoint/llama3-8b-rlhf \
   --micro_train_batch_size 16 \
   --train_batch_size 128 \
   --micro_rollout_batch_size 32 \
   --rollout_batch_size 1024 \
   --max_samples 100000 \
   --max_epochs 1 \
   --prompt_max_len 1024 \
   --generate_max_len 1024 \
   --zero_stage 3 \
   --bf16 \
   --actor_learning_rate 5e-7 \
   --critic_learning_rate 9e-6 \
   --init_kl_coef 0.01 \
   --prompt_data OpenRLHF/prompt-collection-v0.1 \
   --input_key context_messages \
   --apply_chat_template \
   --normalize_reward \
   --packing_samples \
   --adam_offload \
   --flash_attn \
   --gradient_checkpointing \
   --load_checkpoint \
   --use_wandb {wandb_token}
定位卡在train_ppo_ray.py中的185行的ray.get(refs)
https://github.com/OpenRLHF/OpenRLHF/blob/ac2f0a28a7db6df7be1013265e2b534baabec860/openrlhf/cli/train_ppo_ray.py#L185

请问有什么好的解决办法吗

**Comment by hijkzzz** [comment]
用 Hybrid engine

**Comment by lingzehui-123** [comment]
感谢 已解决

**Comment by IlgeeHong** [comment]
Are there any solutions if the hybrid engine is not used?

**Comment by heningsu** [comment]
> 感谢 已解决
想问一下怎么解决的，我也是一模一样的问题，hybrid的脚本方便分享一下吗
